constants/fokabotCommands.py

```diff
@@ -1149,44 +1226,80 @@ async def linkDiscord(fro: str, chan: str, message: list[str]) -> str:
 # @command(
-# async     trigger="!freeze",
-#     privs=privileges.ADMIN_MANAGE_PRIVILEGES,
-#     syntax="<target_name>",
-#     hidden=True,
+#    trigger="!freeze",
+#    privs=privileges.ADMIN_MANAGE_PRIVILEGES,
+#    syntax="<target_name> <reason>",
+#    hidden=True,
 # )
-# def freeze(fro: str, chan: str, message: list[str]) -> str:
-#     """Freeze a specified player."""
-#     target = message[0].lower()
-
-#     if not (target_id := await userUtils.getID(target)):
-#         return "That user does not exist"
-
-#     if await await userUtils.getFreezeTime(target_id):
-#         return "That user is already frozen."
-
-#     await userUtils.freeze(target_id, await userUtils.getID(fro))
-#     return f"Froze {target}."
+# async def freeze(fro: str, chan: str, message: list[str]) -> str:
+#    """Freeze a specified player."""
+#    target = message[0].lower()
+#    reason = " ".join(message[1:])
+
+#    if not (targetID := await userUtils.getID(target)):
+#        return "That user does not exist"
+
+#    if not await userUtils.getFreezeTime(targetID):
+#        return "That user is already frozen."
+
+#    if not reason:
+#        return f"Please specify your reason to freeze {target}."
+
+#    await userUtils.freeze(targetID, await userUtils.getID(fro))
+#    userID = await userUtils.getID(fro)
+#    await rap_logs.send_rap_log(userID, f"has froze {target}")
+#    await rap_logs.send_rap_log_as_discord_webhook(
+#        message="\n".join(
+#            [
+#                f"[{fro}](https://akatsuki.gg/u/{userID}) ({userID}) froze [{target}](https://akatsuki.gg/u/{targetID}).",
+#                f"**Reason**: {reason}",
+#                f"\n> :gear: [View this user](https://old.akatsuki.gg/index.php?p=103&id={targetID}) on **Admin Panel**.",
+#            ],
+#        ),
+#        discord_channel="ac_general",
+#    )
+
+#    await userUtils.appendNotes(targetID, f"{fro} ({userID}) froze {target}")
+#    return f"Froze {target}."
 
 
 @command(
     trigger="!unfreeze",
     privs=privileges.ADMIN_MANAGE_PRIVILEGES,
-    syntax="<target_name>",
+    syntax="<target_name> <reason>",
     hidden=True,
 )
 async def unfreeze(fro: str, chan: str, message: list[str]) -> str:
     """Unfreeze a specified player."""
     target = message[0].lower()
+    reason = " ".join(message[1:])
 
-    if not (target_id := await userUtils.getID(target)):
+    if not (targetID := await userUtils.getID(target)):
         return "That user does not exist"
 
-    if not await userUtils.getFreezeTime(target_id):
+    if not await userUtils.getFreezeTime(targetID):
         return "That user is not frozen."
 
-    await userUtils.unfreeze(target_id, await userUtils.getID(fro))
+    if not reason:
+        return f"Please specify a reason to unfreeze {target}."
+
+    await userUtils.unfreeze(targetID, await userUtils.getID(fro))
+    userID = await userUtils.getID(fro)
```

Comment:
retrieve `userID` first and reuse it in L1288 instead of calling `userUtils.getID()`

each of these getID() calls make a network request to the database

Reply:
noted & fixed in b346c9ca6da9aa4aa864d14e140ae451cf67bbba